fetchai/ledger/api/__init__.py

```diff
@@ -85,10 +85,17 @@ def sync(self, txs: Transactions, timeout=None):
         while True:
             # loop through all the remaining digests and poll them creating a set of completed in this round
             remaining_statuses = [self.tx.status(digest) for digest in remaining]
-            completed_this_round = [status for status in remaining_statuses if status.finished]
-            finished += completed_this_round
 
-            completed_digests = set([status.digest for status in completed_this_round])
+            failed_this_round = [status for status in remaining_statuses if status.failed]
+            if failed_this_round:
+                failures = ['{}:{}'.format(tx_status.digest, tx_status.status) \
+                            for tx_status in failed_this_round]
+                raise RuntimeError('Some transactions have failed: {}'.format(', '.join(failures)))
```

Comment:
👍 👍 👍  - In the future we should make this a distinct exception time I think so that we can attach the status values to this. I am happy to merge this as is however